Include/internal/pycore_runtime.h

```diff
@@ -77,6 +77,11 @@ typedef struct pyruntimestate {
     /* Is Python fully initialized? Set to 1 by Py_Initialize() */
     int initialized;
 
+#if defined(__EMSCRIPTEN__) && defined(PY_CALL_TRAMPOLINE)
+    /* Choose between trampoline based on type reflection vs based on EM_JS */
+    int wasm_type_reflection_available;
```

Comment:
_(also consider using a bool type from stdbool.h instead of int given this seems to be boolean?)_

Reply:
It doesn't need to be publicly accessible. I originally had it as a `static` global in `emscripten_trampoline.c` but I was asked by reviewers to move it here:
https://github.com/python/cpython/pull/106219#discussion_r1253620788
I am perfectly happy to put it back as a `static` global. And yes, it is a boolean I can type it as a `bool`.

Reply:
I much prefer it in runtime state (as you have it) or interpreter state to being a static global.

Reply:
So it doesn't really need to be public but it is so for tidiness reasons. The variables in this struct aren't super public anyways though? I will add a comment here saying which file uses it.

Reply:
Can we put optional bits (and new bits in general) at the end of the struct, instead of near the start? It makes debugging a little easier if the layout doesn't vary as much.